arch/xtensa/core/xtensa-asm2.c

```diff
@@ -61,7 +61,7 @@ void *xtensa_init_stack(struct k_thread *thread, int *stack_top,
 	ret = &bsa[-9];
 
 #ifdef CONFIG_KERNEL_COHERENCE
-	xthal_dcache_region_writeback(ret, (char *)stack_top - (char *)ret);
+	z_xtensa_cache_flush(ret, (char *)stack_top - (char *)ret);
 #endif
 	return ret;
 }
```

Comment:
curiously, this commit (the whole commit, of course, not just this hunk) breaks the synchronization sample app on UP2 ADSP. After this commit I'm getting repeated
```
=== [ADSP Device Reset]
```
messages

Reply:
Can you push a tree? :)  Needless to say this tree as pushed works for me on that sample, it's what I was staring at all yesterday morning.